include/aie/Dialect/AIEVec/IR/AIEVecTypeConstraints.td

```diff
@@ -146,4 +146,14 @@ class IsValidAIE2MatMulShapeAndType<string lhs, string rhs, string acc> :
                                    rhs, VectorOfShapeAndType<[8, 4], BF16>,
                                    acc, VectorOfShapeAndType<[4, 4], F32>>]>>;
 
+class isOperandResultTypePairValidForAIE2MulElem<string lhs, string rhs, string acc> :
+  PredOpTrait<acc # " is not a valid result type for " # 
+              lhs # " x " # rhs # " operands",
```

Comment:
This will print something like:

"result is not a valid result type for lhs x rhs operands"

Which is not very clear, or helpful.

I'm suggesting you to change it to something like:

"'vector<32xi16>' is not a valid result type for 'vector<32xi8>' operands"

You could even go the extra mile and aim for something like:

"'vector<32xi8>' operands produce a 'vector<32xi32>' result, but 'vector<32xi16>' was specified"

These are messages for the user, so the clearer they are, the better. This one requires some extra tablegen to extract the type as a string out of a parameter specified as a string (if you can't figure it out, I'll help). Also, since this is a new error message we're introducing, it should be tested, like in those "invalid.mlir" test files you can see around.